app/models/shipit/undeployed_commit.rb

```diff
@@ -35,12 +35,30 @@ def deploy_disallowed?
     end
 
     def deploy_discouraged?
-      stack.maximum_commits_per_deploy && index >= stack.maximum_commits_per_deploy
+      hit_maximum_commits_per_deploy?
```

Comment:
It's weird to extract the method content, to finally just alias it.

Reply:
I've extracted the method content to re-use it in the `deploy_scheduled?` predicate. In the context of `deploy_scheduled?`, the method name `deploy_discouraged?` does not express the conditions needed to validate the proposition.